spec/sidekiq/terms_of_use/sign_up_service_updater_job_spec.rb

```diff
@@ -184,6 +184,63 @@
       end
     end
 
+    context 'when agreement is changed by ToU acceptance' do
+      let(:response) { 'accepted' }
+      let(:status) { { opt_out: true, agreement_signed: false } }
+
+      before do
+        allow(service_instance).to receive(:agreements_accept)
+        allow(service_instance).to receive(:status).and_return(status)
+      end
+
+      it 'agreement_unchanged returns false' do
+        job.perform(user_account_uuid, version)
+
+        expect(MAP::SignUp::Service).to have_received(:new)
+        expect(service_instance).to have_received(:agreements_accept).with(icn: user_account.icn,
+                                                                           signature_name: common_name,
+                                                                           version:)
+      end
+    end
+
+    context 'when agreement is changed by ToU being declined' do
+      let(:response) { 'declined' }
+      let(:status) { { opt_out: false, agreement_signed: true } }
+
+      before do
+        allow(service_instance).to receive(:agreements_decline)
+        allow(service_instance).to receive(:status).and_return(status)
+      end
+
+      it 'agreement_unchanged returns false' do
+        job.perform(user_account_uuid, version)
+
+        expect(MAP::SignUp::Service).to have_received(:new)
+        expect(service_instance).to have_received(:agreements_accept).with(icn: user_account.icn,
+                                                                           signature_name: common_name,
+                                                                           version:)
+      end
+    end
+
+    context 'when agreement is unchanged' do
+      let(:expected_log) do
+        '[TermsOfUse][SignUpServiceUpdaterJob] Agreement not changed'
+      end
+      let(:status) { { opt_out: false, agreement_signed: true } }
+
+      before do
+        allow(Rails.logger).to receive(:info)
+        allow(service_instance).to receive(:status).and_return(status)
+      end
+
```

Comment:
Can we also add an 'it' block for 'does not make an agreements accept/agreements decline' call? With a test that looks like this:
`expect(service_instance).not_to have_received(:agreements_accept)`